src/action.c

```diff
@@ -912,10 +918,17 @@ actions_run(struct view *activator, struct server *server,
 				view_move_to_edge(view, edge, snap_to_windows);
 			}
 			break;
+		case ACTION_TYPE_TOGGLE_SNAP_TO_EDGE:
 		case ACTION_TYPE_SNAP_TO_EDGE:
 			if (view) {
 				/* Config parsing makes sure that direction is a valid direction */
 				enum view_edge edge = action_get_int(action, "direction", 0);
+				if (action->type == ACTION_TYPE_TOGGLE_SNAP_TO_EDGE
+						&& view->tiled == edge) {
+					view_set_untiled(view);
+					view_restore_to(view, view->natural_geometry);
```

Comment:
Can we end up without a valid `natural_geometry` here?

Reply:
Sorry - can't remember. @jlindgren90 @ahesford - do you know off the top of your heads?

Reply:
natural_geometry can be empty for xdg views, but I believe this is correct anyway - the 0x0 gets passed to the client to choose its own size and then we center the view in the commit handler.